src/app/upgrade.cpp

```diff
@@ -37,6 +37,22 @@
 
 namespace
 {
+    void storeLegacyDefaults()
+    {
+        const QVariantHash legacyDefaults {
+            {QLatin1String {"BitTorrent/Session/QueueingSystemEnabled"}, true}
+        };
+
+        SettingsStorage *settingsStorage {SettingsStorage::instance()};
+        for (auto it = legacyDefaults.cbegin(); it != legacyDefaults.cend(); ++it) {
+            const QString &name = it.key();
+            const QVariant &legacyDefault = it.value();
+            const QVariant currentValue = settingsStorage->loadValue(name, legacyDefault);
+            if (currentValue == legacyDefault)
```

Comment:
If the following is correct, I think it is more straightforward to express it:
```c++
const QVariant currentValue = settingsStorage->loadValue(name);
if (currentValue.isNull())
    settingsStorage->storeValue(name, legacyDefault);
```

Reply:
Wow, good catch! :+1:
I'll change it.